ServerCore/Pages/Events/FastestSolves.cshtml.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Events
+{
+    public class FastestSolvesModel : EventSpecificPageModel
+    {
+        private readonly ServerCore.Models.PuzzleServerContext _context;
+
+        public List<PuzzleStats> Puzzles { get; private set; }
+
+        public FastestSolvesModel(ServerCore.Models.PuzzleServerContext context)
+        {
+            _context = context;
+        }
+
+        public async Task OnGetAsync()
+        {
+            var puzzlesData = await PuzzleStateHelper.GetSparseQuery(_context, this.Event, null, null)
+                .Where(s => s.SolvedTime != null && s.Puzzle.IsPuzzle)
+                .GroupBy(state => state.Puzzle)
+                .Select(g => new {
+                    Puzzle = g.Key,
+                    SolveCount = g.Count(),
+                    Fastest = g.OrderBy(s => s.SolvedTime - s.UnlockedTime).Take(3).Select(s => new { s.Team.ID, s.Team.Name, Time = s.SolvedTime - s.UnlockedTime})
+                })
+                .OrderByDescending(p => p.SolveCount).ThenBy(p => p.Puzzle.Name)
+                .ToListAsync();
+
+            var puzzles = new List<PuzzleStats>(puzzlesData.Count);
+            for (int i = 0; i < puzzlesData.Count; i++)
+            {
+                var data = puzzlesData[i];
+                var stats = new PuzzleStats()
+                {
+                    Puzzle = data.Puzzle,
+                    SolveCount = data.SolveCount,
+                    SortOrder = i
+                };
+
+                var fastEnum = data.Fastest.GetEnumerator();
```

Comment:
Is there a reason to use GetEnumerator() instead of a for loop? C# best practices typically recommend considering the Enumerator as an internal implementation detail whenever possible.

Reply:
If you switch to a for loop over a collection it would also let us change the number of fastest solves we want to display in the future. Anything that we can leave flexible is likely to simplify future changes.

Reply:
Replaced it with a version that probably uses more memory but uses the "best practices" APIs.